operator/v1/types_network.go

```diff
@@ -428,7 +430,22 @@ type HybridOverlayConfig struct {
 	HybridOverlayVXLANPort *uint32 `json:"hybridOverlayVXLANPort,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="self == oldSelf || has(self.mode)",message="ipsecConfig.mode is required"
 type IPsecConfig struct {
+	// mode defines the behaviour of the ipsec configuration within the platform.
+	// Valid values are `Disabled`, `External`, `Full` and omitted.
+	// When 'Disabled', ipsec will not be enabled at the node level.
+	// When 'External', ipsec is enabled on the node level but requires the user to configure the secure communication parameters.
+	// This mode is for external secure communications and the configuration can be done using the k8s-nmstate operator.
+	// When 'Full', ipsec is configured on the node level and inter-pod secure communication within the cluster is configured.
+	// Note with `Full`, if ipsec is desired for communication with external (to the cluster) entities (such as storage arrays), 
+	// this is left to the user to configure.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default which is
+	// subject to change over time.
+	// The current default behaviour is `Full`.
+	// +kubebuilder:validation:Enum=Disabled;External;Full
```

Comment:
Spoke with David, since our ratchet makes this field effectively required, we shouldn't mention the omitted case.
```suggestion
	// mode defines the behaviour of the ipsec configuration within the platform.
	// Valid values are `Disabled`, `External`,  and `Full`.
	// When 'Disabled', ipsec will not be enabled at the node level.
	// When 'External', ipsec is enabled on the node level but requires the user to configure the secure communication parameters.
	// This mode is for external secure communications and the configuration can be done using the k8s-nmstate operator.
	// When 'Full', ipsec is configured on the node level and inter-pod secure communication within the cluster is configured.
	// Note with `Full`, if ipsec is desired for communication with external (to the cluster) entities (such as storage arrays), 
	// this is left to the user to configure.
	// +kubebuilder:validation:Enum=Disabled;External;Full
```